eng/pipelines/templates/stages/archetype-sdk-client.yml

```diff
@@ -77,7 +77,11 @@ stages:
           - ${{ each config in parameters.AdditionalMatrixConfigs }}:
             -  ${{ config }}
         MatrixFilters: ${{ parameters.MatrixFilters }}
-        MatrixReplace: ${{ parameters.MatrixReplace }}
+        MatrixReplace:
```

Comment:
@benbp does this seem reasonable from a matrix replace scenario?

Reply:
I can't think of a better way to do it at the moment.

It does bring up an idea though, right now we have "first class" support for the `container` field in a matrix, meaning the pipeline yaml knows to look for that specific field and inject it via the template as a yaml parameter. We could do a similar thing with `condition` statements, e.g.

```
{
  "matrix": {
    "BuildConfigurationGroup": {
      "Debug": {
        "BuildConfiguration": "Debug",
        "condition": "ne(variables['System.TeamProject'], 'internal')"
      },
      "Release": {
        "BuildConfiguration": "Release",
        "condition": "eq(variables['System.TeamProject'], 'internal')"
      }
    }
  }
}
```

And then in the job yaml:

```
condition: $(condition)
```

In this specific case, I don't think it would be appropriate, because you'd have twice as many jobs in the pipeline UI with half showing as skipped, but just something to keep in mind as an option for the future should it make sense.